App/Sagas/TextileSagas.ts

```diff
@@ -611,8 +576,9 @@ export function * showImagePicker(action: ActionType<typeof UIActions.showImageP
         width: pickerResponse.width,
         isVertical: pickerResponse.isVertical
       }
-      yield put(CameraRollActions.imagePickerSuccess(threadId, image))
-      yield call(NavigationService.navigate, 'SharePhoto', {threadId, image})
+      yield put(UIActions.updateSharingPhotoThread(threadId))
+      yield put(UIActions.updateSharingPhotoImage(image))
+      yield call(NavigationService.navigate, 'ThreadSharePhoto', { backTo: 'ViewThread' })
```

Comment:
The AddComment screen can be canceled out of by tapping the back button, but depending on what context you're in (adding from the wallet screen or adding from the thread detail screen), you need to go back to a different place. To support that we pass in a route name as `backTo` that the AddComment screen can ready and use when the back button is tapped.